docs/proposals/21-ipv6-singlestack-local.md

```diff
@@ -0,0 +1,289 @@
+For example, when using IPv6 single-stack networking on cloud infrastructure, users might want to use IPv6 prefixes assigned to them by the provider instead of configuring the shoot's node and pod CIDRs upfront.
+
+While focusing on IPv6 single-stack networking for now, we cannot neglect how a future IPv4/IPv6 dual-stack implementation may look like and provide a corresponding outlook.
+However, these ideas only serve an informational purpose for motivating decisions regarding the IPv6 single-stack implementation.
+Once this enhancement has been implemented and IPv6 single-stack networking in local shoots is a stable feature, further changes can be proposed to support IPv6 single-stack on cloud infrastructure, and eventually IPv4/IPv6 dual-stack networking.
+
+### Goals
+
+- augment all relevant Gardener API types defined in gardener/gardener to allow selecting either IPv4 or IPv6 single-stack networking (`core.gardener.cloud/v1beta1.{Shoot,Seed}`, `extensions.gardener.cloud/v1alpha1.{Network,DNSRecord}`)
+- define a contract that all components (including networking extensions) need to follow to support this configuration
+- adapt all relevant components in the local Gardener environment to support this configuration
+- document all required steps for getting started with the IPv6 setup for developers (without requiring existing config/knowledge)
+- add e2e tests to prevent regressions and guarantee the stability of the feature (especially because nobody is running it in productive environments)
+
+### Non-Goals
+
+- support IPv4/IPv6 dual-stack networking
+- provide IPv4 connectivity on IPv6 single-stack clusters
+- support IPv6 single-stack networking on cloud infrastructure
+- use IPv6 prefixes assigned by provider for pod CIDR similar to [provider-assigned node CIDR](https://github.com/gardener/gardener/blob/219d828fcdea81fb3edf13de2736daf81e137923/pkg/operation/botanist/infrastructure.go#L73)
+- support changing the networking setup of shoots from IPv4 to IPv6 single-stack or vice-versa
+- host IPv6 single-stack shoots on IPv4 single-stack seeds or vice-versa
+- propose changes API types defined outside of gardener/gardener and their implementations, e.g., the `NetworkConfig` APIs (`providerConfig`) of networking extensions
+- support the legacy VPN tunnel (`ReversedVPN` feature gate is disabled)
+
+## Proposal
+
+A new feature gate `IPv6SingleStack` is added to gardener-apiserver.
+The IPv6-related fields and values can only be used if the feature gate is enabled.
+The feature gate cannot be enabled if the `ReversedVPN` feature gate is disabled (the legacy VPN tunnel solution is [not supported by provider-local](https://github.com/gardener/gardener/blob/83de074f1bc1c009f92e97a08289340591377af6/docs/extensions/provider-local.md#current-limitations) anyway).
```

Comment:
Drop this in the light of above PR to prevent it being outdated immediately. Probably drop everything about `ReversedVPN` in this doc.

Reply:
Yes, you are right. Removed the lines.